Documentation/planning/multi-sdk-band-support.md

```diff
@@ -0,0 +1,258 @@
+4. **Prepare** 2xx bootstrap inputs - `<PSB 1xx artifacts> + <CSB 1xx artifacts>`
+5. **Checkout** 2xx branch
+6. **Build** using `./build.sh --with-artifacts <2xx bootstrap inputs>`
+7. **Prepare** 2xx inputs - `<PSB 1xx artifacts> + <CSB 1xx artifacts> + <CSB 2xx artifacts>`
+8. **Build** using `./build.sh --with-sdk <CSB 2xx SDK> --with-artifacts <2xx inputs>`
+
+### A distro maintainer wishing to support only a 2xx SDK (monthly workflow)
+
+A distro maintainer who only wants to support the 2xx SDK still needs to build the 1xx branch to obtain the runtime and an updated 1xx SDK for use the next month. In this case, we build the 2xx inputs with the 2xx SDK.
+
+1. **Checkout** 1xx branch:
+2. **Build** using `./build.sh --with-sdk <PSB 1xx SDK> --with-artifacts <PSB 1xx artifacts>`.
+3. **Gather** 1xx build outputs from `artifacts/x64/Release` = `<CSB 1xx artifacts>`.
+4. **Prepare** 2xx inputs - `<PSB 2xx artifacts> + <CSB 1xx artifacts> + <PSB 1xx artifacts>`
+5. **Checkout** 2xx branch
+6. **Build** using `./build.sh --with-sdk <PSB 2xx SDK> --with-artifacts <2xx inputs>`
+
+### A distro maintainer wishing to support all available SDKs (monthly workflow)
+
+A distro maintainer wishing to support all SDK bands builds the 1xx band, then may build all Nxx bands in parallel.
+
+1. **Checkout** 1xx branch:
+2. **Build** using `./build.sh --with-sdk <PSB 1xx SDK> --with-artifacts <PSB 1xx artifacts>`.
+3. **Gather** 1xx build outputs from `artifacts/x64/Release` = `<CSB 1xx artifacts>`.
+
+The following additional steps may be done in parallel:
+
+For each SDK band Nxx:
+
+   1. **Prepare** Nxx inputs - `<PSB Nxx artifacts> + <CSB 1xx artifacts> + <PSB 1xx artifacts>`
+   2. **Checkout** Nxx branch
+   3. **Build** using `./build.sh --with-sdk <PSB Nxx SDK> --with-artifacts <Nxx inputs>`
+
+## Poison and Prebuilt detection
+
+Prebuilt detection should not change. Poisoning workflows may need to change. We don't want to poison inputs from the 1xx VMR, used when building the 2xx VMR. We **do** want to poison the previously source built inputs prior to adding in the 1xx inputs. So, in a poisoning check on a 2xx VMR build, the following workflow would be used:
+
+1. **Checkout** 1xx branch:
+2. **Build** using `./build.sh --with-sdk <PSB 1xx SDK> --with-artifacts <PSB 1xx artifacts>`.
+3. **Gather** 1xx build outputs from `artifacts/x64/Release` = `<CSB 1xx artifacts>`.
+4. **Prepare (pre-poison)** 2xx inputs - `<PSB 2xx artifacts> + <PSB 1xx artifacts>`
+5. **Poison** 2xx inputs
+6. **Augment (post-poison)** 2xx inputs - `<2xx inputs> <CSB 1xx artifacts>`
+7. **Checkout** 2xx branch
+8. **Build** using `./build.sh --with-sdk <PSB 2xx SDK> --with-artifacts <2xx inputs>`
+
+## Potential issues and mitigations
+
+There are a few potential issues that may require tweaks to this approach, source fixes, additional guidance, etc. They are listed below:
+
+### A newer SDK band may continue to require an older (typically 1xx band) SDK to build
```

Comment:
I would like to a policy in place that prevent this situation.

Reply:
Should the policy say that "A newer band SDK **may** successfully build with an older band, but should **always** succesfully build with the newer band"?

Reply:
That's my vote.